### DISCUSSION TITLE
resolution option in addIterativeLSI() and addClusters()

### DISCUSSION THREAD

**Discussion by janekim319** [GreenleafLab/ArchR#946] [Questions / Documentation] [answered]
Could someone explain how the resolution option in addIterativeLSI() and addClusters() are different? I think the resolution in addClusters() directly affects the number of clusters, but I'm unsure the effect that addIterativeLSI's resolution has on the number of clusters.

Below are the example uses of those two functions:

projHeme2 <- addIterativeLSI(
    ArchRProj = projHeme2,
    useMatrix = "TileMatrix", 
    name = "IterativeLSI", 
    iterations = 2, 
    clusterParams = list( #See Seurat::FindClusters
        **resolution = c(0.2),** 
        sampleCells = 10000, 
        n.start = 10
    ), 
    varFeatures = 25000, 
    dimsToUse = 1:30
)

projHeme2 <- addClusters(
    input = projHeme2,
    reducedDims = "IterativeLSI",
    method = "Seurat",
    name = "Clusters",
    **resolution = 0.8**
)

How does using 0.2 in addIterativeLSI and 0.8 in addCluster affect the clustering results?

**Comment by rcorces** [comment, marked as answer]
Might be helpful to review how Iterative LSI works in ArchR:
https://www.archrproject.com/bookdown/iterative-latent-semantic-indexing-lsi.html

There is a clustering step that helps in identifying variable features but those clusters are not stored, they are just used during the LSI process. Then you go an add clusters based on your final dim reduction.

**Reply by janekim319** [reply]
Thank you so much!

**Reply by janekim319** [reply]
Hi Ryan, I have a follow up question. From my understanding ArchR uses Seurat's FindNeighbors() function which uses SNN. However, I'm not seeing a resolution parameter that can be passed into FindNeighbors(). Is the resolution parameter something for addIterativeLSI? It would be super helpful to know which Seurat function addIterativeLSI uses! Thank you!

**Reply by rcorces** [reply]
For questions like this, please read the function documentation pages in depth. Almost always, the information is already there.
`addIterativeLSI()` has param `clusterParams`
![image](https://user-images.githubusercontent.com/10343046/127547140-bcaac994-bb87-4368-993e-270c2fab040d.png)
which is passed to `addClusters()`
If you look at the `addClusters()` page (https://www.archrproject.com/reference/addClusters.html), you'll see that the three-dots additional arguments are passed through to `Seurat::FindClusters()`

We've spent a **lot** of time on the documentation and appreciate every effort you can make to find answers on your own.
